### PR TITLE
[414] remove NetworkSelector on sc wallet session

### LINKED ISSUE #414: Uni merge: Network selector is enabled when connected to SC wallets
Originally reported in https://github.com/cowprotocol/cowswap/pull/406#issuecomment-1098807170 (case 2)

Network selector is enabled when connected to WC wallets (as an example, to Safe). So when I try to change a network, I see this warning message, that seems to me illogical as Safe wallet does not support networks changing. So can we disable this dropdown when connected to SC wallets at least?

### PR DESCRIPTION
**Pull request by W3stside**
Closes #414 

image

## Test
1. connect SC wallet
2. no net selector

### CODE REVIEW

File: src/custom/components/Header/NetworkSelector/NetworkSelectorMod.tsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ import { UnsupportedChainIdError, useWeb3React } from 'web3-react-core'
 import { useAddPopup, useRemovePopup } from 'state/application/hooks'
 import { useEffect } from 'react'
 import { getExplorerBaseUrl } from 'utils/explorer'
+import { useWalletInfo } from '@src/custom/hooks/useWalletInfo'
```
Comment (alfetopito) [inline src/custom/components/Header/NetworkSelector/NetworkSelectorMod.tsx:33]: Path police!

```suggestion
import { useWalletInfo } from 'hooks/useWalletInfo'
```

**Review by alfetopito — APPROVED** [review]

**Comment by elena-zh** [comment]
Gnosis Safe in Rinkeby: 
image

Gnosis safe in GC: 
image

**Review by elena-zh — DISMISSED** [review]
See the notes above

**Comment by elena-zh** [comment]
Also, there is no network selector to a not connected user: 
image

**Comment by elena-zh** [comment]
Network selector presents when connect to Gnosis Safe mobile wallet (all networks)
image

**Comment by W3stside** [comment]
Elena can you check this is the case outside of this PR?

**Comment by elena-zh** [comment]
Hey @W3stside , now the network switcher is displayed to a not connected user. And the network switcher is removed when connected to Gnosis Safe in Mainnet, and to a mobile Gnosis Safe wallet in Mainnet. 

But still, the selector is displayed when connected to **Argent wallet**
argent

Gnosis Safe in Gnosis Chain 
GC

Gnosis Safe in Rinkeby 
rink

**Comment by W3stside** [comment]
@elena-zh yeah didnt have the right changes in there, could u recheck after the build is remade?

**Comment by alfetopito** [comment]
@W3stside 
Screen Shot 2022-05-04 at 11 41 11

**Comment by elena-zh** [comment]
Hey @W3stside , works nicely now with Gnosis Safe wallet for all networks. 

However, I still see the network selector when connected to Argent or Pillar wallets. Could we hide it? 

Thanks

**Comment by W3stside** [comment]
@elena-zh pillar i wont do rn since i have no experience with it but argent should be done now

**Comment by elena-zh** [comment]
@W3stside , seems that there is an issue with the latest build

**Comment by W3stside** [comment]
> @W3stside , seems that there is an issue with the latest build

there wasn't, what do you see wrong?

**Comment by elena-zh** [comment]
Argent still shows the network switcher
image

**Comment by W3stside** [comment]
> Argent still shows the network switcher image

hmm it doesn't do that for me
image

if it comes up again let's hotfix

**Comment by elena-zh** [comment]
@W3stside , I opened a separate low-prio issue #526 for this case, as the issue is still reproducible to me.

**Pull request merged**